### DISCUSSION TITLE
How do I enable Fluent Bit logging in EKS Fargate without running sidecar containers?

### DISCUSSION THREAD

**Discussion by autero1** [gruntwork-io/knowledge-base#282] [Help] [answered]
Does anyone have a concrete example of how to enable Fluent Bit logging for Fargate pods using the [eks-cluster](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/modules/services/eks-cluster) and [k8s-service](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/modules/services/k8s-service) Terraform modules from the Gruntwork  Service Catalog?

I'd like to do this without running a sidecar container, which appears to be supported but I'm struggling with how and where to pass through the required configuration in Fargate-land.

**Comment by autero1** [comment, marked as answer]
You can set up Fluent Bit for Fargate using the [`eks-fargate-container-logs`](https://github.com/gruntwork-io/terraform-aws-eks/tree/master/modules/eks-fargate-container-logs) module. For a concrete example of using the module, see the [EKS Fargate Cluster With Supporting Service](https://github.com/gruntwork-io/terraform-aws-eks/tree/master/examples/eks-fargate-cluster-with-supporting-services/core-services). For full configuration options and available destinations, see [`variables.tf`](https://github.com/gruntwork-io/terraform-aws-eks/blob/master/modules/eks-fargate-container-logs/variables.tf).

You don't need to apply any special configuration to your `k8s-service` - Fluent Bit will forward your container logs to the configured destination automatically.

**Reply by yorinasub17** [reply]
Also as a follow up, since version [0.60.0](https://github.com/gruntwork-io/terraform-aws-service-catalog/releases/tag/v0.60.0) of the service catalog, the `eks-core-services` module will setup log aggregation for Fargate pods using the `eks-fargate-container-logs` module.

**Reply by jalessio** [reply]
@yorinasub17 I tracked down the issue. While the `eks-core-services` module does indeed have `enable_fargate_fluent_bit` defaulting to `true` [code](https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/c28247f0e1eb79ac521b8c4fe1f06f3cc914c245/modules/services/eks-core-services/variables.tf#L64-L68), the example provided with the Gruntwork-deployed Reference Architecture did not set a value for `fargate_fluent_bit_execution_iam_role_arns` which defaults to an empty list [code](https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/c28247f0e1eb79ac521b8c4fe1f06f3cc914c245/modules/services/eks-core-services/variables.tf#L168-L172), thus no logs were actually making it to CloudWatch. 

In `_envcommon/services/eks-core-services.hcl` I set this variable, which got the logs flowing:

```
fargate_fluent_bit_execution_iam_role_arns = [dependency.eks_cluster.outputs.eks_default_fargate_execution_role_arn]
```

I think the Gruntwork Reference Architecture deployed for customers needs to be updated to reflect this missing IAM role or other customers will find themselves in the same situation.

**Reply by yorinasub17** [reply]
Ah you are right! That is indeed a bug in the deployment. Have prioritized to get this fixed, and the examples should be updated once it flows through.
